app/models/claims/training_allowance.rb

```diff
@@ -0,0 +1,54 @@
+class Claims::TrainingAllowance
+  def initialize(mentor:, provider:, academic_year:, claim_to_exclude: nil)
+    @mentor = mentor
+    @provider = provider
+    @academic_year = academic_year
+    @claim_to_exclude = claim_to_exclude
+  end
+
+  def training_type
+    has_initial_training? ? REFRESHER_TRAINING : INITIAL_TRAINING
+  end
+
+  def total_hours
+    training_type == INITIAL_TRAINING ? INITIAL_TRAINING_HOURS : REFRESHER_TRAINING_HOURS
+  end
+
+  def remaining_hours
+    total_hours - hours_completed
+  end
```

Comment:
Likewise, this method also performs an expensive query.

```suggestion
  def remaining_hours
    @remaining_hours ||= total_hours - hours_completed
  end
```